### PR TITLE
use of container representation should not change container headers

### PR DESCRIPTION
**Pull request by bourgeoa**
resolves Issue #1387

### CODE REVIEW

File: test/integration/http-test.js [hunk 1]
```diff
@@ -167,8 +167,7 @@ describe('HTTP APIs', function () {
         .end(done)
     })
 
-    // This test is probably wrong: it is not a container if there is an index page
-    it.skip('should have set Link as resource on a implicit index page', function (done) {
+    it('should have set Link as Container/BasicContainer on a implicit index page', function (done) {
```
Comment (TallTed) [inline test/integration/http-test.js:170]:
```suggestion
    it('should have set Link as Container/BasicContainer on an implicit index page', function (done) {
```

**Review by jeff-zucker — COMMENTED** [review]
This makes sense for URLs in the form https://user.example.com/ but not for https://example.com/.  The second one is not a pim:Storage, is it?

**Comment by jeff-zucker** [comment]
Also, wouldn't this violate the semantics of pim:Storage?  AFAIK you can never have one Storage inside another Storage.  If the server root is a Storage than all the user Storages are inside it.

**Comment by bourgeoa** [comment]
> This makes sense for URLs in the form https://user.example.com/ but not for https://example.com/. The second one is not a pim:Storage, is it?

Agreed I opened an issue. I did not had a solution to find the provider Uri.

> Also, wouldn't this violate the semantics of pim:Storage? AFAIK you can never have one Storage inside another Storage. If the server root is a Storage than all the user Storages are inside it.

No because https://user.example.com/ is not a container of https://example.com

**Comment by bourgeoa** [comment]
@jeff-zucker  Anyway this PR is not about pim:storage, but only container representation. See files changed

**Comment by jeff-zucker** [comment]
> @jeff-zucker Anyway this PR is not about pim:storage, but only container representation. See files changed

Then I am afraid I do not understand the PR at all. What do you mean by container representation?  Doesn't this line mark the root folder as pim:Storage?
```
if (req.path === '/' && req.app.locals.host.serverUri.contains(req.hostname)) fileMetadata.isStorage = true
```

**Comment by bourgeoa** [comment]
The resulting PR is only one line and a test

image

**Comment by bourgeoa** [comment]
merged with PR #1646

**Pull request closed without merging**